managed/CounterStrikeSharp.API/Modules/Menu/CenterHtmlMenu.cs

```diff
@@ -13,114 +13,73 @@
         {
             _plugin = plugin;
-            RemoveOnTickListener();
-            plugin.RegisterListener<Core.Listeners.OnTick>(Display);
+            _onTick = DisplayCenterHtmlMenu;
+            _plugin.RegisterListener(_onTick);
         }
-        
-        public override void Display()
+
+        private void DisplayCenterHtmlMenu()
         {
-            if (MenuManager.GetActiveMenu(Player) != this)
-            {
-                Reset();
-                return;
-            }
-            
-            var builder = new StringBuilder();
-            builder.Append($"<b><font color='yellow'>{Menu.Title}</font></b>");
-            builder.AppendLine("<br>");
+            Player.PrintToCenterHtml(_message);
+        }
 
+        public override void Display()
+        {
             var keyOffset = 1;
-
-            for (var i = CurrentOffset; i < Math.Min(CurrentOffset + MenuItemsPerPage, Menu.MenuOptions.Count); i++)
+            var builder = new StringBuilder();
+            builder.Append($"{Menu.Title}<br>");
+            for (var i = CurrentOffset;
+                 i < Math.Min(CurrentOffset + MenuItemsPerPage, Menu.MenuOptions.Count);
+                 i++)
             {
                 var option = Menu.MenuOptions[i];
-                string color = option.Disabled ? "grey" : "green";
-                builder.Append($"<font color='{color}'>!{keyOffset++}</font> {option.Text}");
-                builder.AppendLine("<br>");
+
+                builder.Append(
+                    $"<font color='{(option.Disabled ? "gray" : "green")}'>[!{keyOffset++}]</font> {option.Text}<br>");
             }
-            
+
             if (HasPrevButton)
             {
-                builder.AppendFormat("<font color='yellow'>!7</font> &#60;- Prev");
-                builder.AppendLine("<br>");
+                builder.Append($"<font color='yellow'>[!7] -></font> Prev<br>");
             }
 
             if (HasNextButton)
             {
-                builder.AppendFormat("<font color='yellow'>!8</font> -> Next");
-                builder.AppendLine("<br>");
+                builder.Append($"<font color='yellow'>[!8] -></font> Next<br>");
             }
 
-            builder.AppendFormat("<font color='red'>!9</font> -> Close");
-            builder.AppendLine("<br>");
+            builder.Append($"<font color='red'>[!9] -></font> Exit<br>");
 
-            var currentPageText = builder.ToString();
-            Player.PrintToCenterHtml(currentPageText);
+            _message = builder.ToString();
         }
 
         public override void Reset()
         {
-            base.Reset();
-            RemoveOnTickListener();
-            
-            // Send a blank message to clear the menu
-            Player.PrintToCenterHtml(" ");
-        }
-
-        private void RemoveOnTickListener()
-        {
-            var onTick = new Core.Listeners.OnTick(Display);
-            _plugin.RemoveListener("OnTick", onTick);
```

Comment:
The changes you have made here aren't really needed, I feel like they add ambiguity.